stripe/src/main/java/com/stripe/android/AbstractEphemeralKey.java

```diff
@@ -216,7 +216,7 @@ String getType() {
     protected static <TEphemeralKey extends AbstractEphemeralKey> TEphemeralKey
     fromString(@Nullable String rawJson, Class ephemeralKeyClass) throws JSONException {
         if (rawJson == null) {
-            return null;
+            throw new IllegalArgumentException("Exception instantiating " + ephemeralKeyClass+ " null raw key");
```

Comment:
How about `"Attempted to instantiate " + ephemeralKeyClass + " with null key JSON"`? Can you also add a test? Does it need to be `ephemeralKeyClass.getSimpleName()`?

Reply:
`getSimpleName` makes the name a tad shorter, since the error message is quite long.

I'll change the text

A test is not trivial because of the was `TestEphemeralKeyProvider` is build, but I'll re-try

Reply:
all done by the way